.github/workflows/ci.yaml

```diff
@@ -110,109 +110,94 @@ jobs:
     outputs:
       labels: ${{ steps.prep.outputs.docker_image_labels }}
       version: ${{ steps.prep.outputs.docker_image_tag }}
-      golinkerargs: ${{ steps.prep.outputs.go_linker_args }}
-      cgo_cflags: ${{ steps.prep.outputs.cgo_cflags }}
-      goversion: ${{ steps.prepenv.outputs.goversion }}
+      go-linker-args: ${{ steps.prep.outputs.go_linker_args }}
+      cgo-cflags: ${{ steps.prep.outputs.cgo_cflags }}
+      go-version: ${{ steps.prepenv.outputs.goversion }}
       registry: ${{ steps.prepenv.outputs.registry }}
       repository: ${{ steps.prepenv.outputs.repository }}
 
   build-amd64:
     name: Build amd64 docker image
-    needs: [prepare]
-    uses: ./.github/workflows/build-dockerimage.yaml
-    with:
-      platform: amd64
-      goversion: ${{ needs.prepare.outputs.goversion }}
-      golinkerargs: ${{ needs.prepare.outputs.golinkerargs }}
-      cgo_cflags: ${{ needs.prepare.outputs.cgo_cflags }}
-      labels: ${{ needs.prepare.outputs.labels }}
-      image-tag: ${{ needs.prepare.outputs.version }}
+    needs: [prepare, tests, linting]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: build-image
+        uses: ./.github/actions/build-image
+        with:
+          platform: amd64
+          go-version: ${{ needs.prepare.outputs.go-version }}
+          go-linker-args: ${{ needs.prepare.outputs.go-linker-args }}
+          cgo-cflags: ${{ needs.prepare.outputs.cgo-cflags }}
+          labels: ${{ needs.prepare.outputs.labels }}
+          image-tag: ${{ needs.prepare.outputs.version }}
+      - name: upload-image
```

Comment:
name should be "Upload Image", if an id is required that should be set separately